libmamba/src/core/transaction.cpp

```diff
@@ -1449,17 +1439,16 @@ namespace mamba
         };
 
         int mode = SOLVER_TRANSACTION_SHOW_OBSOLETES | SOLVER_TRANSACTION_OBSOLETE_IS_UPGRADE;
-        transaction_classify(m_transaction, mode, &classes);
-        Id cls;
-        for (int i = 0; i < classes.count; i += 4)
+        transaction_classify(m_transaction, mode, classes.get());
+        auto const cls_end = classes.end();
+        for (auto cls_iter = classes.begin(); cls_iter < cls_end; cls_iter += 4)
```

Comment:
is it really better to use iterators here? I feel like it would be safer to use the index instead of going through an iterator (potentially we could do bounds checking with the indedx function, and it's more straightforward).

Reply:
Changed back to index and `ObjQueue::at`.